### PR TITLE
Fix duplicate scrapping by Prometheus

### PR DESCRIPTION
**Pull request by TheoBrigitte**
Towards: https://github.com/giantswarm/giantswarm/issues/25216

This PR fixes an issue where node exporter was scrapped/monitored twice by Prometheus.

This is due to mutually exclusive jobs in Prometheus and the missing `giantswarm.io/monitoring` annotation on pod resources.

In more details Prometheus has 
- node-exporter job: old job we are moving away from
  it selects pods by name and namespace
  it ignores **pod** with the `giantswarm.io/monitoring` label or annotation; this condition exists so we can use the following job

- workload job: new job used for most things
  it selects endpoint which have the `giantswarm.io/monitoring` label or annotation on their **service**

In our case we have:
- node-exporter service with the `giantswarm.io/monitoring` annotation ([link](https://github.com/giantswarm/node-exporter-app/blob/301f0db16eb0a635c1e977ebacb388dad3e4e804/helm/node-exporter-app/values.yaml#L13-L22)) => this will be added into `workload` job
- pod missing the `giantswarm.io/monitoring` annotation => this will **also** be added into the `node-exporter` job

therefore we are scrapping node-exporter twice.

tl;dr adding the `giantswarm.io/monitoring` annotation to pods, make sure it is correctly ignored by the old `node-exporter` job in Prometheus

### CODE REVIEW

**Comment by TheoBrigitte** [comment]
This was tested in https://github.com/giantswarm/prometheus-meta-operator/pull/1112#issuecomment-1373629110

**Comment by TheoBrigitte** [comment]
### Before

```yaml
$ ks get po -lapp.kubernetes.io/name=node-exporter  -oyaml|yq -y -r '.items[].metadata.annotations'
cluster-autoscaler.kubernetes.io/safe-to-evict: 'true'
---
cluster-autoscaler.kubernetes.io/safe-to-evict: 'true'
---
cluster-autoscaler.kubernetes.io/safe-to-evict: 'true'
---
cluster-autoscaler.kubernetes.io/safe-to-evict: 'true'
---
cluster-autoscaler.kubernetes.io/safe-to-evict: 'true'
---
cluster-autoscaler.kubernetes.io/safe-to-evict: 'true'
```

### After

```yaml
$ ks get po -lapp.kubernetes.io/name=node-exporter  -oyaml|yq -y -r '.items[].metadata.annotations'
cluster-autoscaler.kubernetes.io/safe-to-evict: 'true'
giantswarm.io/monitoring: 'true'
prometheus.io/scrape: 'true'
---
cluster-autoscaler.kubernetes.io/safe-to-evict: 'true'
giantswarm.io/monitoring: 'true'
prometheus.io/scrape: 'true'
---
cluster-autoscaler.kubernetes.io/safe-to-evict: 'true'
giantswarm.io/monitoring: 'true'
prometheus.io/scrape: 'true'
---
cluster-autoscaler.kubernetes.io/safe-to-evict: 'true'
giantswarm.io/monitoring: 'true'
prometheus.io/scrape: 'true'
---
cluster-autoscaler.kubernetes.io/safe-to-evict: 'true'
giantswarm.io/monitoring: 'true'
prometheus.io/scrape: 'true'
---
cluster-autoscaler.kubernetes.io/safe-to-evict: 'true'
giantswarm.io/monitoring: 'true'
prometheus.io/scrape: 'true'
```

**Review by QuantumEnigmaa — APPROVED** [review]

**Review by hervenicol — APPROVED** [review]
Quite counter-intuitive, can't wait for proper pod or service-monitors all around!
